### ISSUE TITLE
S3 object keys are prepended with forward slash

### ISSUE THREAD

**Issue by adam-fowler** [localstack/localstack#2140]
The following awscli command does not work when used with localstack
```
aws --endpoint-url http://localhost:4572 s3 ls s3://my-bucket/
```
It only ever returns one entry
```
                           PRE /
```
This appears to be caused by the fact that the awscli is using the "prefix" attribute. I can use my web browser to see the contents using `http://localhost:4572/my-bucket`.

The most likely cause of this is localstack prepends object keys with "/" which is inconsistent with AWS. This means localstack requires prefixes to start with "/" while AWS doesn't.

This also causes issues with the delimiter field. If I want to list all the files at the root folder of an S3 bucket I should be able to use `http://localhost:4572/my-bucket?delimiter=/`. This returns nothing probably because the first character of every file is "/".

**Comment by adam-fowler** [comment]
More details. I found this only occurs if I call putObject with a key that includes a leading forward slash. It looks like AWS automatically removes this while localstack doesn't. If I don't include the leading forward slash everything works as expected.

**Comment by jgbmattos** [comment]
Hi @adam-fowler.
I was looking for that issue and couldn't reproduce your problem. The same behaviour is seen in AWS and Localstack in my  case.
If your filename starts with a slash s3 simulates a folder and in order for you access that files in that folder you need to specify that, for example, in your case, http://localhost:4572/my-bucket//.
AWS
`aws s3 ls s3://a81c7377-ae1c-43b1-b2e9-5d79431dd955/ 
                           PRE /
                           PRE teste/
2020-03-16 09:50:22     252662 evidencia-1.png

aws s3 ls s3://a81c7377-ae1c-43b1-b2e9-5d79431dd955//
2020-03-16 09:53:11     252662 evidencia-1.png
`
Localstack

`aws --endpoint-url http://localhost:4572 s3 ls s3://my-bucket/ 
                           PRE /
                           PRE teste/
2020-03-16 09:51:12     252662 evidenci-a.png

aws --endpoint-url http://localhost:4572 s3 ls s3://my-bucket//
2020-03-16 09:53:40     252662 evidenci-a.png
`

**Comment by whummer** [comment]
Thanks for looking into this @adam-fowler @jgbmattos . I also wasn't able to reproduce this issue, the commands provided by @jgbmattos are working for me, too.

Closing this issue as "cannot reproduce", but please do re-open it with specific instructions to reproduce if the problem persists:
* Version of Python, 'aws'; please make sure to test with LocalStack `latest` Docker image 
* Exact sequence of bash commands (using `aws` or `awslocal`) to reproduce the issue end-to-end.

Thanks!

**Comment by adam-fowler** [comment]
@jgbmattos @whummer Sorry I took so long to respond. I also have to apologise as the issue is my fault. 

I am working on an AWS SDK for the swift programming language [aws-sdk-swift](https://github.com/swift-aws/aws-sdk-swift) and the difference I was seeing between localstack and AWS was due to a difference in the way I process endpoint urls and the AWS virtual hosts. The virtual host processing was clearing the "/" at the front of the key. When working with localstack the virtual host addressing is disabled so it produced a different result.
